### DISCUSSION TITLE
Sort only the import types without rearranging the lines

### DISCUSSION THREAD

**Discussion by Alc-Alc** [astral-sh/ruff#12310] [Q&A] [answered]
Is there a ruff config to do this?
- sort the individual types
- do not sort the lines

Before
```py
from .yyy import BBB, AAA
from .xxx import DDD, CCC
```

After (Notice how only the types are sorted, not the lines)
```py
from .yyy import AAA, BBB
from .xxx import CCC, DDD
```

Current (I do not want this behavior)

```py
from .xxx import CCC, DDD
from .yyy import AAA, BBB
```

Reason: I use this in `__init__.py` and if I change the order, it causes a circular import

Thanks for your work on the project.

**Comment by MichaReiser** [comment]
HI @Alc-Alc no option disables sorting of the imported modules entirely. Can you tell me more about how you determine the order of your import statements or do you want to allow arbitrary import ordering?

**Reply by Alc-Alc** [reply]
Thank you for responding @MichaReiser. 🙂

I apologize in advance for this rather long explanation. I am being verbose in hopes I explain my usecase properly.

This is how one of my "sub packages" looks like.

```
users                
├── __init__.py     
├── models.py
├── routes.py                              
└── service.py
```

- `models.py` does not depend on the other modules here in this sub package.
- `service.py` depends on `models.py`.
- `routes.py` depends on both `service.py` and `models.py`.

This is how `__init__.py` looks like.

```py
from .models import ModelOne, ModelTwo
from .service import ServiceOne, ServiceTwo
from .routes import RouterOne
```

If I let ruff sort, it changes it to (note how routes.py is moved before service.py) and this is what causes the `ImportError`.

```py
from .models import ModelOne, ModelTwo
from .routes import RouterOne
from .service import ServiceOne, ServiceTwo
```

>  do you want to allow arbitrary import ordering

I want to know if there is a way I can tell ruff on a per file / glob basis, "respect the order of lines in this file, but you can sort the imported classes / functions however you see fit"

If you feel this is too niche of a usecase or plainly "your project setup is messed up" that is a perfectly OK response I will be contended with. 🙂

**Reply by MichaReiser** [reply, marked as answer]
Thanks for the detailed explanation. That helped!

I assume that this exception only applies to a few imports? If so, you can tell Ruff to not sort a specific import by using `isort: skip`

https://docs.astral.sh/ruff/linter/#action-comments

**Reply by Alc-Alc** [reply]
Yes, this is applicable only to my `__init__.py` files. Thanks. I will turn off ruff sort and resort to manual sorting.
